### PR TITLE
Restart all the PostgreSQL nodes after Add/Remove command

### PR DESCRIPTION
**Pull request by deenchev**
### :nut_and_bolt: Description: What code changed, and why?

Added PostgreSQL nodes restart logic in the Add/Remove flow, in order to apply the newest HAProxy configuration. Currently the configuration is being updated, but due to the `reconfigure` hook we introduced for HAProxy, the service isn't getting restarted and is not using the latest configuration (thus routing traffic to outdated nodes). That's why we want to restart the habitat supervisor of all the PostgreSQL nodes, if a PostgreSQL node/s is added/removed.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-backend-haproxy/habitat/config/haproxy.conf [hunk 1]
```diff
@@ -20,10 +20,10 @@ listen postgresql
     default-server inter 3s fall 3 rise 2 on-marked-down shutdown-sessions
 {{~#if bind.database}}
   {{~#eachAlive bind.database.members as |dbmember|}}
-    server {{dbmember.sys.ip}} {{dbmember.sys.ip}}:{{dbmember.cfg.port}} maxconn {{cfg.server.maxconn}} check port "${PGLEADERCHK_PORT}"
+    server {{dbmember.sys.ip}} {{dbmember.sys.ip}}:{{dbmember.cfg.port}} maxconn {{cfg.server.maxconn}} check port 6432
   {{~/eachAlive}}
 {{~else}}
-    server {{sys.ip}} {{sys.ip}}:{{cfg.pg.port}} maxconn {{cfg.server.maxconn}} check port "${PGLEADERCHK_PORT}"
+    server {{sys.ip}} {{sys.ip}}:{{cfg.pg.port}} maxconn {{cfg.server.maxconn}} check port 6432
```
Comment (punitmundra) [inline components/automate-backend-haproxy/habitat/config/haproxy.conf:26]: we can use this 
6432 => {{cfg.pgleaderchk.port}}

**Review by jayvikramsharma1 — APPROVED** [review]

File: components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go [hunk 2]
```diff
@@ -929,3 +930,18 @@ func removeRestrictedKeysFromSrcFile(srcString string) (string, error) {
...
 	}
 }
+
+func (nu *NodeUtilsImpl) restartHabSupOnBackend(service string) error {
+	remoteExcecutor := NewRemoteCmdExecutorWithoutNodeMap(&SSHUtilImpl{}, cli.NewWriter(os.Stdout, os.Stderr, os.Stdin))
+	infra, _, err := nu.getHaInfraDetails()
+	if err != nil {
+		return err
+	}
+	flags := &RestartCmdFlags{
```
Comment (jayvikramsharma1) [inline components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go:940]: not handling the service, use it like 

```
flags := &RestartCmdFlags{}
switch service {
     case POSTGRESQL:
          flags.postgresql = true
     case OPENSEARCH: 
         flags.opensearch = true
}

Comment (deenchev) [reply components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go]: Done.

File: components/automate-backend-haproxy/habitat/config/haproxy.conf (hunk 1 above)
Comment (deenchev) [reply components/automate-backend-haproxy/habitat/config/haproxy.conf]: Done.

**Review by punitmundra — APPROVED** [review]

**Review by punitmundra — APPROVED** [review]

**Pull request merged**